vxquery-core/src/main/java/org/apache/vxquery/compiler/rewriter/rules/util/ExpressionToolbox.java

```diff
@@ -197,8 +197,11 @@ public static void getConstantAsPointable(ConstantExpression typeExpression, Tag
     }
 
     public static int getTypeExpressionTypeArgument(Mutable<ILogicalExpression> searchM) {
-        final int ARG_TYPE = 1;
+        int ARG_TYPE = 1;
         AbstractFunctionCallExpression searchFunction = (AbstractFunctionCallExpression) searchM.getValue();
+        if (searchFunction.getFunctionIdentifier().equals(BuiltinFunctions.FN_COUNT_1.getFunctionIdentifier())) {
```

Comment:
It seems like this may be used out of context and that is why we have a condition for count. Why do you need and if statement for count?